### PR TITLE
Add southbound API to pass notification to RackHD (refer to ODR682)

### PR DESCRIPTION
**Pull request by cgx027**
This is the PR to support https://github.com/RackHD/on-http/pull/377

### CODE REVIEW

**Comment by yyscamper** [comment]
@tannoa2 This is the notification API that you want.

**Comment by iceiilin** [comment]
Looks great to me

**Comment by yyscamper** [comment]
Please change the wait-completion-uri job as well: https://github.com/RackHD/on-tasks/blob/master/lib/jobs/wait-completion-uri.js#L57

File: lib/jobs/install-os.js [hunk 1]
```diff
@@ -236,6 +237,18 @@ function installOsJobFactory(
                     }
                 }
             });
+
+            self._subscribeTaskNotification(function(data) {
+                assert.object(data);
+                if(
+                    data.nodeId === self.nodeId
+                    && data.taskId === self.taskId
+                    && data.type === 'osInstall'
```
Comment (yyscamper) [inline lib/jobs/install-os.js]: I have some concern to judge the `type` here.
- the taskId is enough to tell this is install os job;
- this makes  the code cannot be shared by other job that also wants notification, for example, for wait-completion-uri job, to define another type "waitCompletionUri"? I think it's not worth.

**Comment by cgx027** [comment]
Remove WIP flag.

**Comment by WangWinson** [comment]
:+1:

**Comment by yyscamper** [comment]
:+1:

**Comment by yyscamper** [comment]
jenkins: test this please

**Pull request merged**